### PR TITLE
[MI-2473] Create custom post for subscription notifications and share…

### PR DESCRIPTION
**Pull request by ayusht2810**
… record post

1. Create custom posts for shared record posts and subscription notifications.
2. Fix the issue of the comment modal and update state modal opening on multiple clients.

image
image

### CODE REVIEW

File: webapp/src/containers/custom_posts/notificationPost.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                    <span>
+                        {field.value}
+                    </span>
+                </td>,
+            );
+
+            rowPos += 1;
+        });
+
+        if (headerCols.length > 0) {
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/custom_posts/notificationPost.tsx:76]:
```suggestion
        if (headerCols.length) {
```

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: webapp/src/containers/custom_posts/notificationPost.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        recordId: record_id,
+        recordType: record_type,
+    };
+
+    const getNotificationBody = (): JSX.Element => {
+        const fieldTables = [] as JSX.Element[];
+        let headerCols = [] as JSX.Element[];
+        let bodyCols = [] as JSX.Element[];
+        let rowPos = 0;
+        let nrTables = 0;
```
[the remark this answers is not in the document]
Comment (ayusht2810) [reply webapp/src/containers/custom_posts/notificationPost.tsx:24]: this is used as a unique key for the table element

[the remark this answers is not in the document]
Comment (ayusht2810) [reply webapp/src/containers/custom_posts/notificationPost.tsx:24]: no

**Review by raghavaggarwal2308 — APPROVED** [review]

File: webapp/src/plugin_constants/index.ts [hunk 3]
```diff
@@ -146,23 +146,31 @@ export enum RecordDataLabelConfigKey {
     ASSIGNMENT_GROUP = 'assignment_group',
```
Comment (manojmalik20) [inline webapp/src/plugin_constants/index.ts:146]: The name of this enum can be `RecordDataConfigKeys`

File: webapp/src/utils/index.tsx [hunk 4]
```diff
@@ -77,7 +77,12 @@ export const validateKeysContainingLink = (key: string) => (
     key === KnowledgeRecordDataLabelConfigKey.AUTHOR ||
     key === KnowledgeRecordDataLabelConfigKey.CATEGORY ||
     key === RecordDataLabelConfigKey.ASSIGNED_TO ||
-    key === RecordDataLabelConfigKey.ASSIGNMENT_GROUP
+    key === RecordDataLabelConfigKey.ASSIGNMENT_GROUP ||
```
Comment (manojmalik20) [inline webapp/src/utils/index.tsx:80]: You can create a `Set` for this as well.

File: webapp/src/containers/addOrViewComments/index.tsx [hunk 5]
```diff
@@ -38,11 +38,14 @@ const AddOrViewComments = () => {
...
         setRefetch(false);
+        setShowErrorPanel(false);
     }, []);
 
     const hideModal = useCallback(() => {
         dispatch(resetGlobalModalState());
-        resetFieldStates();
+        setTimeout(() => {
+            resetFieldStates();
+        }, 500);
```
[the remark this answers is not in the document]
Comment (ayusht2810) [reply webapp/src/containers/addOrViewComments/index.tsx]: removed this time now

File: webapp/src/containers/addOrViewComments/index.tsx [hunk 6]
```diff
@@ -38,11 +38,14 @@ const AddOrViewComments = () => {
         setApiError(null);
         setValidationError('');
         setRefetch(false);
+        setShowErrorPanel(false);
     }, []);
 
     const hideModal = useCallback(() => {
         dispatch(resetGlobalModalState());
-        resetFieldStates();
+        setTimeout(() => {
```
[the remark this answers is not in the document]
Comment (ayusht2810) [reply webapp/src/containers/addOrViewComments/index.tsx:46]: I don't think we need that as we have removed time from it

File: webapp/src/containers/updateState/index.tsx [hunk 7]
```diff
@@ -38,7 +38,9 @@ const UpdateState = () => {
 
     const hideModal = useCallback(() => {
         dispatch(resetGlobalModalState());
-        resetStates();
+        setTimeout(() => {
```
[the remark this answers is not in the document]
Comment (ayusht2810) [reply webapp/src/containers/updateState/index.tsx:41]: same

**Review by manojmalik20 — APPROVED** [review]

File: webapp/src/containers/custom_post/styles.scss [hunk 8]
```diff
@@ -0,0 +1,53 @@
+.servicenow-post {
+    .shared-post {
+        border: 1px solid var(--center-channel-color-40);
+        border-left-style: solid;
+        border-left-width: 4px;
```
Comment (sooraj-shukla) [inline webapp/src/containers/custom_post/styles.scss:5]: border-left: 4px solid rgba(35, 137, 215, 0.5);

File: webapp/src/containers/custom_post/styles.scss [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+        border-radius: 0 4px 4px 0;
+        margin: 5px 0;
+        padding: 12px;
+        border-left-color: rgba(35, 137, 215, 0.5);
```
Comment (sooraj-shukla) [inline webapp/src/containers/custom_post/styles.scss:9]:
```suggestion
```

File: webapp/src/containers/custom_post/styles.scss [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+        background: rgb(var(--center-channel-bg-rgb));
+
+        &__pretext {
+            font-size: 13.5px;
+        }
+
+        &__title {
+            font-weight: 600;
```
Comment (sooraj-shukla) [inline webapp/src/containers/custom_post/styles.scss:17]: Please check if we've util classes for font weights.

**Review by sooraj-shukla — CHANGES_REQUESTED** [review]

**Review by sooraj-shukla — APPROVED** [review]

**Pull request merged**